src/caffeine/caffeine.c

```diff
@@ -90,6 +93,15 @@ void caf_deallocate(mspace heap, void* mem)
   mspace_free(heap, mem);
 }
 
+// take address in a segment and convert to an address on given image
+intptr_t caf_convert_base_addr(void* addr, int image)
+{
+   ptrdiff_t offset = (byte*)addr - (byte*)gex_Segment_QueryAddr(mysegment);
+   void* segment_start_remote_image = NULL;
+   gex_Event_Wait(gex_EP_QueryBoundSegmentNB(myworldteam, image - 1, &segment_start_remote_image, NULL, NULL, 0));
+   return (intptr_t)((char *)segment_start_remote_image + offset);
```

Comment:
If we are adopting the convention of using `byte *` for pointer arithmetic in C, then please apply it consistently throughout this file.
```suggestion
   return (intptr_t)((byte *)segment_start_remote_image + offset);
```
There may be other occurences..

Reply:
@bonachea Thanks. After fixing this, the only other occurrence of a type cast with `char *` or `char*` is in the collectives, but no pointer math is being done on the resulting pointer.